### PR TITLE
boards: shields: Add Seeed Studio XIAO Expansion Board

### PR DESCRIPTION
**Pull request by CharlesDias**
Add Seeed Studio XIAO Expansion Board:
- User button
- SSD1306 128x64 pixels
- MicroSD
- PFC8563 RTC

### CODE REVIEW

File: boards/shields/seeed_xiao_expansion_board/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+.. _seeed_xiao_expansion_board:
+
+Seed Studio XIAO Expansion Board
```
Comment (kartben) [inline boards/shields/seeed_xiao_expansion_board/doc/index.rst:3]: Needs s/Seed/Seeed/ in the entire changeset

File: boards/shields/seeed_xiao_expansion_board/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+################################
+
+Overview
+********
+
+Seeed Studio XIAO Expansion Board is a powerful functional expansion board
+for `Seeed Studio XIAO series`_ of only half Raspberry Pi 4 size. It enables
+building prototypes and projects in an easy and quick way. With its rich
+peripherals, you could explore the infinite possibilities of Seeed Studio
+XIAO series. Circuit python is also well supported by this board.
```
Comment (kartben) [inline boards/shields/seeed_xiao_expansion_board/doc/index.rst:13]:
```suggestion
XIAO series.
```

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/shields/seeed_xiao_expansion_board/doc/index.rst (hunk 1 above)
Comment (CharlesDias) [reply boards/shields/seeed_xiao_expansion_board/doc/index.rst]: Done.

File: boards/shields/seeed_xiao_expansion_board/doc/index.rst (hunk 2 above)
Comment (CharlesDias) [reply boards/shields/seeed_xiao_expansion_board/doc/index.rst]: Done. Thanks for your review!

**Comment by CharlesDias** [comment]
Reduced image size.

**Comment by kartben** [comment]
> Reduced image size.

Thanks! I just pushed a fixup commit where the image is now transparent, which is even better IMO
also removed the pins diagram as I don't think it adds a lot of value vs. people going to Seeed Studio wiki/website if they need more info.

If that works for you, I'll let you squash the fixup commit :)

**Review by kartben — APPROVED** [review]
very cool!

**Comment by CharlesDias** [comment]
> Thanks! I just pushed a fixup commit where the image is now transparent, which is even better IMO also removed the pins diagram as I don't think it adds a lot of value vs. people going to Seeed Studio wiki/website if they need more info.
> 
> If that works for you, I'll let you squash the fixup commit :)

Thanks, @kartben! It's perfect for me. Just one question. Now I have to squash the fixup commit and push force? Thanks

**Comment by kartben** [comment]
> > Thanks! I just pushed a fixup commit where the image is now transparent, which is even better IMO also removed the pins diagram as I don't think it adds a lot of value vs. people going to Seeed Studio wiki/website if they need more info.
> > If that works for you, I'll let you squash the fixup commit :)
> 
> Thanks, @kartben! It's perfect for me. Just one question. Now I have to squash the fixup commit and push force? Thanks

Yes!

**Comment by CharlesDias** [comment]
Update, @kartben. Thank you for your help!

**Review by erwango — APPROVED** [review]

**Pull request merged**